src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -325,6 +338,59 @@ public DictionaryEntriesSearchThread(final SourceTextEntry newEntry) {
         }
     }
 
+    /**
+     * Thread for user requested dictionary search.
+     */
+    public class DictionaryTextSearchThread extends Thread {
```

Comment:
Is there a reason you didn't extend `EntryInfoSearchThread`? It handles error logging and edge cases like when the current entry has changed before the results come back.

Reply:
(See `DictionaryEntriesSearchThread` for an example)